torch/fx/node.py

```diff
@@ -439,6 +439,20 @@ def is_impure(self):
 
         return False
 
+    def replace_input_with(self, old_input: 'Node', new_input: 'Noe'):
+        """
+        Replace input node with new input node.
+        """
+        def maybe_replace_node(n : Node) -> Node:
+            return new_input if n == old_input else n
+
+        new_args = map_arg(self.args, maybe_replace_node)
+        new_kwargs = map_arg(self.kwargs, maybe_replace_node)
+        assert isinstance(new_args, tuple)
+        assert isinstance(new_kwargs, dict)
+        self.__update_args_kwargs(new_args, new_kwargs)
```

Comment:
Overall I'm fine with having this method, however I'd like to understand the UX in terms of discoverability of the `map_arg` pattern. How hard is it to know to use `map_arg`? Do you think that should be a method on `Node` that applies to both `args` and `kwargs`? (essentially a generalization of this function)

Reply:
> How hard is it to know to use map_arg

I think ```map_arg``` is pretty easy to use and powerful and I am happy with that :)